Source/Scene/Expression.js

```diff
@@ -56,8 +56,11 @@ define([
 
     });
 
-    Expression.prototype.evaluate = function(feature) {
-        return this._runtimeAst.evaluate(feature);
+    Expression.prototype.evaluate = function(feature, result) {
```

Comment:
OK, so it would be used as an optimization if we are expecting to have to evaluate colors.

So for the case I mentioned, would I just check if the result parameter is defined during eval, and if not, create the scratch variable then?